From 4e4e0d7e24b86e36cc9030b31b70c5e7bbffed89 Mon Sep 17 00:00:00 2001
From: Takamasa Kuramitsu <takamasa.kuramitsu@sony.com>
Date: Tue, 5 Dec 2017 10:55:44 +0900
Subject: [PATCH] Avoid crash on Files app when creating a directory

Files app crashes if a client document provider creates a file
in createDocument callback unexpectedly during the directory
creation sequence.

This CL fixes to avoid the crash on Files app and show a Snackbar
message instead.

diff --git a/src/com/android/documentsui/CreateDirectoryFragment.java b/src/com/android/documentsui/CreateDirectoryFragment.java
index 66be663d..e7a33a09 100644
--- a/src/com/android/documentsui/CreateDirectoryFragment.java
+++ b/src/com/android/documentsui/CreateDirectoryFragment.java
@@ -141,7 +141,8 @@ protected DocumentInfo doInBackground(Void... params) {
                         resolver, mCwd.derivedUri.getAuthority());
                 final Uri childUri = DocumentsContract.createDocument(
                         client, mCwd.derivedUri, Document.MIME_TYPE_DIR, mDisplayName);
-                return DocumentInfo.fromUri(resolver, childUri);
+                DocumentInfo doc = DocumentInfo.fromUri(resolver, childUri);
+                return doc.isDirectory() ? doc : null;
             } catch (Exception e) {
                 Log.w(TAG, "Failed to create directory", e);
                 return null;
